packages/keyring-controller/src/KeyringController.ts

```diff
@@ -2155,6 +2155,10 @@ export class KeyringController extends BaseController<
     for (const serializedKeyring of serializedKeyrings) {
       await this.#restoreKeyring(serializedKeyring);
     }
+
+    if (this.#keyrings.length !== this.#keyringsMetadata.length) {
```

Comment:
Interesting. Previously this was not checked until vault update, correct? So a mismatch here previously would allow the user to log in, but any actions triggering vault failure would fail.

With this check added here, we detect the corruption sooner. Maybe less risk of other operations failing in a confusing (potentially harmful) way, but has the downside of locking the user out completely if the failure occurs. 

Probably this is a good call. We'll soon have the vault recovery flow as backup in case this does happen to a user somehow.

Reply:
Exactly! I found this issue when adding a test case for too many objects in metadata.

> We'll soon have the vault recovery flow as backup in case this does happen to a user somehow
That would be awesome. My doubt was indeed because of locking out users, but I thought that letting them continue the execution would pose a greater risk

Reply:
Good point. We can make an issue for this, and treat it as a blocker for ever dropping support for a keyring (which maybe we'll never intentionally want to do anyway)